oidb/pepper/service/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+    let PepperRequest {
+        jwt,
+        overriding_aud,
+        epk_hex_string,
+        epk_expiry_time_secs,
+        epk_blinder_hex_string,
+        uid_key,
+    } = request;
+
+    let claims = aptos_oidb_pepper_common::jwt::parse(jwt.as_str())
+        .map_err(|e| anyhow!("JWT decoding error: {e}"))?;
+
+    let actual_uid_key = if let Some(uid_key) = uid_key.as_ref() {
+        uid_key
+    } else {
+        "sub"
+    };
+
+    let uid_val = if actual_uid_key == "email" {
+        claims
+            .claims
+            .email
+            .clone()
+            .ok_or_else(|| anyhow!("`email` required but not found in jwt"))?
+    } else if actual_uid_key == "sub" {
+        claims.claims.sub.clone()
+    } else {
+        bail!("unsupported uid key: {}", actual_uid_key)
+    };
+
+    let blinder = hex::decode(epk_blinder_hex_string)
+        .map_err(|e| anyhow!("blinder unhexlification error: {e}"))?;
+    let epk_bytes = hex::decode(epk_hex_string)
+        .map_err(|e| anyhow!("epk unhexlification error: {e}"))?;
+    let epk = bcs::from_bytes::<EphemeralPublicKey>(&epk_bytes)
+        .map_err(|e| anyhow!("epk bcs deserialization error: {e}"))?;
+    let recalculated_nonce = OpenIdSig::reconstruct_oauth_nonce(
+        blinder.as_slice(),
+epk_expiry_time_secs,
+        &epk,
+        &Configuration::new_for_devnet(),
+    )
+    .map_err(|e| anyhow!("nonce reconstruction error: {e}"))?;
+
+    ensure!(
+        claims.claims.nonce == recalculated_nonce,
+        "with nonce mismatch"
+    );
+
+    let key_id = claims
+        .header
+        .kid
+        .ok_or_else(|| anyhow!("missing kid in JWT"))?;
+
+    let sig_pub_key = jwk::cached_decoding_key(&claims.claims.iss, &key_id)?;
+    let mut validation_with_sig_verification = Validation::new(RS256);
+    validation_with_sig_verification.validate_exp = false; // Don't validate the exp time
+    let _claims = jsonwebtoken::decode::<Claims>(
+        jwt.as_str(),
+        sig_pub_key.as_ref(),
+        &validation_with_sig_verification,
+    ) // Signature verification happens here.
+    .map_err(|e| anyhow!("JWT signature verification failed: {e}"))?;
+
+    // Decide the client_id in the input.
+    let actual_aud = if ACCOUNT_DISCOVERY_CLIENTS.contains(&claims.claims.aud) {
+        if let Some(aud) = overriding_aud.as_ref() {
+            aud
+        } else {
+            &claims.claims.aud
+        }
+    } else {
+        &claims.claims.aud
+    };
+
+    let input = PepperInput {
+        iss: claims.claims.iss.clone(),
+        uid_key: actual_uid_key.to_owned(),
+        uid_val,
+        aud: actual_aud.clone(),
+    };
+    let input_bytes = bcs::to_bytes(&input).unwrap();
+    let (pepper, vuf_proof) = vuf::scheme0::Scheme0::eval(&VUF_SCHEME0_SK, &input_bytes)?;
+    ensure!(vuf_proof.is_empty(), "internal proof error");
+    let pepper_hexlified = hex::encode(pepper);
+    Ok(pepper_hexlified)
```

Comment:
Why are you not returning the proof here? @heliuchuan will want to verify the proof in the SDK.

Reply:
because the proof is empty?

Reply:
For BLS, sure, but when you change the scheme, will you put the proof inside `pepper_hexlified`?

Reply:
in that case, pepper should be `bcs::to_bytes(some_struct_that_include_output_and_proof)`?

Reply:
Sure, but the problem with this PR is that we do not have clean separation between structs and their serialization.

Why is `PepperResponse` containing hex strings? It should have `Vec<u8>`'s and the serialization code (BCS, JSON) will decide the format.

Reply:
fixed